libraries/standard/mqtt/include/mqtt_lightweight.h

```diff
@@ -47,76 +48,215 @@ typedef struct MqttPublishInfo MQTTPublishInfo_t;
 {
-    MQTTSuccess = 0,
-    MQTTBadParameter,
-    MQTTNoMemory,
-    MQTTSendFailed,
-    MQTTRecvFailed,
-    MQTTBadResponse,
-    MQTTServerRefused
+    MQTTSuccess = 0,    /**< Function completed successfully. */
+    MQTTBadParameter,   /**< At least one parameter was invalid. */
+    MQTTNoMemory,       /**< A provided buffer was too small. */
+    MQTTSendFailed,     /**< The transport send function failed. */
+    MQTTRecvFailed,     /**< The transport receive function failed. */
+    MQTTBadResponse,    /**< An invalid packet was received from the server. */
+    MQTTServerRefused   /**< The server refused a CONNECT or SUBSCRIBE. */
 } MQTTStatus_t;
 
+/**
+ * @brief MQTT Quality of Service values.
+ */
 typedef enum MQTTQoS
 {
-    MQTTQoS0 = 0,
-    MQTTQoS1 = 1,
-    MQTTQoS2 = 2
+    MQTTQoS0 = 0,    /**< Delivery at most once. */
+    MQTTQoS1 = 1,    /**< Delivery at least once. */
+    MQTTQoS2 = 2     /**< Delivery exactly once. */
 } MQTTQoS_t;
 
+/**
+ * @brief Buffer passed to MQTT library.
+ *
+ * These buffers are not copied and must remain in scope for the duration of the
+ * MQTT operation.
+ */
 struct MQTTFixedBuffer
 {
-    uint8_t * pBuffer;
-    size_t size;
+    uint8_t * pBuffer;    /**< @brief Pointer to buffer. */
+    size_t size;          /**< @brief Size of buffer. */
 };
 
+/**
+ * @brief MQTT CONNECT packet parameters.
+ */
 struct MQTTConnectInfo
 {
+    /**
+     * @brief Whether to establish a new, clean session or resume a previous session.
+     */
     bool cleanSession;
+
+    /**
+     * @brief MQTT keep alive period.
+     */
     uint16_t keepAliveSeconds;
+
+    /**
+     * @brief MQTT client identifier. Must be unique per client.
+     */
     const char * pClientIdentifier;
+
+    /**
+     * @brief Length of the client identifier.
+     */
     uint16_t clientIdentifierLength;
+
+    /**
+     * @brief MQTT user name. Set to NULL if not used.
+     */
     const char * pUserName;
+
+    /**
+     * @brief Length of MQTT user name. Set to 0 if not used.
+     */
     uint16_t userNameLength;
+
+    /**
+     * @brief MQTT password. Set to NULL if not used.
+     */
     const char * pPassword;
+
+    /**
+     * @brief Length of MQTT password. Set to 0 if not used.
+     */
     uint16_t passwordLength;
 };
 
+/**
+ * @brief MQTT SUBSCRIBE packet parameters.
+ */
 struct MQTTSubscribeInfo
 {
+    /**
+     * @brief Quality of Service for subscription.
+     */
     MQTTQoS_t qos;
+
+    /**
+     * @brief Topic filter for subscription.
+     */
     const char * pTopicFilter;
+
+    /**
+     * @brief Length of subscription topic filter.
+     */
     uint16_t topicFilterLength;
 };
 
+/**
+ * @brief MQTT PUBLISH packet parameters.
+ */
 struct MqttPublishInfo
 {
+    /**
+     * @brief Quality of Service for message.
+     */
     MQTTQoS_t qos;
+
+    /**
+     * @brief Whether this is a retained message.
+     */
     bool retain;
+
+    /**
+     * @brief Topic name for message.
```

Comment:
Topic name on which the message is published.

Reply:
changed